qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -975,16 +976,17 @@ public void testRecovery() throws Exception {
                     if (false == line.startsWith("p")) {
                         continue;
                     }
-                    Matcher m = Pattern.compile("(\\d+\\.\\d+\\.\\d+)$").matcher(line);
+                    Matcher m = Pattern.compile("((\\d+)\\.\\d+\\.\\d+)$").matcher(line);
                     assertTrue(line, m.find());
                     String version = m.group(1);
+                    int major = Integer.parseInt(m.group(2));
                     if (currentLuceneVersion.equals(version)) {
                         numCurrentVersion++;
                     } else if (bwcLuceneVersion.equals(version)) {
                         numBwcVersion++;
-                    } else if (minCompatibleBWCVersion.equals(version) && minCompatibleBWCVersion.equals(bwcLuceneVersion) == false) {
-                        // Our upgrade path from 7.non-last always goes through 7.last, which depending on timing can create 7.last
-                        // index segment. We ignore those.
+                    } else if (major == currentLuceneVersionMajor - 1) {
```

Comment:
This is a less-precise check than previously, but segment versions no longer have minor-version semver information on them